spec/features/openid_connect/openid_connect_spec.rb

```diff
@@ -569,6 +569,26 @@
     expect(userinfo_response[:verified_at]).to be_nil
   end
 
+  it 'returns the locale if requested', driver: :mobile_rack_test do
+    user = user_with_2fa
+
+    token_response = sign_in_get_token_response(
+      user: user,
+      scope: 'openid email locale',
+    )
+
+    access_token = token_response[:access_token]
+    expect(access_token).to be_present
+
+    page.driver.get api_openid_connect_userinfo_path,
+                    {},
+                    'HTTP_AUTHORIZATION' => "Bearer #{access_token}"
+
+    userinfo_response = JSON.parse(page.body).with_indifferent_access
+    expect(userinfo_response[:email]).to eq(user.email)
+    expect(userinfo_response[:locale]).to eq('en')
```

Comment:
would it be a more illustrative test if we did a non-default (non-en) locale?

Reply:
I've been working on this one for a bit and it is surprisingly difficult. The helpers that we use to work through the sign in and redirect process all assume the default locale. Some of them do unhelpful things like explicitly visit an path without a locale param which resets the locale to `en`.

I'll keep hacking on it to see what I come up with but it is getting messy.

Reply:
I pushed f1e8a38fde417f807deba72e41b0fb3136d8cd88 to the other branch. It has what I came up with. I'll move it over here.